microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc

```diff
@@ -0,0 +1,109 @@
+
+Once an identity has been established according to the steps described by the MicroProfile JWT specification the remaining security decisions required are covered by the relevant Jakarta EE specifications in use by the various subsystems. 
+
+=== Nice-to-Have Requirements
+
+It will not be a priority of this enhancement to add support for alternative authentication mechanisms to `MP-JWT` as this brings a lot of additional complexity.  For this initial integration how we provide support for the `MP-JWT` mechanism should be largely transparent to the deployment allowing us to evolve this aspect further.
+
+Some of the issues we need to consider may be worthwhile discussing at the specification level.
+
+Presently SmallRye JWT is making use of EE Security to provide an authentication mechanism.  A big benefit of making use of EE Security is that a deployment can also make use of this specification to provide an `@IdentityStore` implementation as described in the MP JWT specification.  Deployments also have the option of bringing in their own authentication mechanisms if they are not making use of `MP-JWT`.
+
+However there are a couple of draw backs from the EE Security approach.  A first drawback is this brings in dependencies on JASPI and JACC both of which must be correctly enabled on the server, the `MP-JWT` mechanism is a very trivial light weight mechanism so a lot of overhead is pulled in.  A second draw back is that it should be possible for standard and vendor specific mechanisms to be specified, this brings us additional challenges as many of our mechanisms are native WildFly Elytron mechanisms.  Making use of native mechanisms has it's own complexities as we may need to bridge into using an `@IdentityStore` made available by CDI, alternatively this may require configuration of domains not covered by MicroProfile Config values.  We could consider porting some of our vendor specific mechanisms to EE security but one of the features of our mechanisms is the ability to support multiple mechanisms concurrently which is not completely covered by EE security.  The `MP-JWT` mechanism is also ideally suited to being used in parallel with other authentication mechanisms however we could end up in a situation with a hybrid of native mechanisms and an EE Security mechanism.
+
+Overall these issues can be considered and solved, however I believe the primary motivation for activating MicroProfile JWT is to make use of the `MP-JWT` authentication mechanism so that will be the primary focus of this RFE, depending on time constraints consideration of alternative authentication mechanisms will either take place towards the end of this enhancement or in a follow up enhancement.
```

Comment:
Yes after further review and commencing work on the integration it makes no sense to avoid merging this RFE whilst solving alternative authentication mechanisms, it also does not make sense to deffer the complete RFE to a later WildFly release to support alternative authentication mechanisms so that should be discussed on it's own.

We will track ongoing work and towards the end of this RFE start to decide where it the correct home for ongoing work,

Reply:
@darranl Do I understand that you want to decide during work on this RFE where will the enhancement regarding alternative authentication mechanisms end up? As @rsvoboda said - this RFE needs to have clear scope. This decision needs to be included in this document.

Reply:
No that discussion is entirely outside the scope of this RFE, this RFE is purely about providing support for the MicroProfile JWT specification with the MP-JWT authentication mechanism only.

All further planning will be outside this RFE.

Reply:
+1 will change that - that now is 100% in follow and and TBH I want to discuss that at the spec level first.

Reply:
@honza-kasik I have removed that sentence entirely, this block of text is now in the non-requirements section but that sentence came down when it was still in the nice to have section.